### ISSUE TITLE
Deploy workload during vcluster creation

### ISSUE THREAD

**Issue by agracey** [loft-sh/vcluster#302]
It would be helpful for people building multi-tenancy on top of vcluster to be able to deploy a workload in the same step as the vcluster creation. 

My use case is that I would like to build an operator that deploys a vcluster and some workload on top of it. The problem that I'm facing is that I need to switch my kubeconfig from the service account to one that's provided mid-script. While this is likely not too hard if I shell out to work with the vcluster cli, I would much rather have a way to pass through rendered yaml to the vcluster helm chart.

**Comment by FabianKramm** [comment]
@agracey thanks for creating this issue! We are currently working on a way to create your own plugins for vcluster, which one of those use cases would be to deploy custom manifests or helm charts on vcluster startup, similar to how we do it ourselves with core dns.

**Comment by agracey** [comment]
Thank you for the quick reply! I think for my use case, either method would be pretty useful and I'm interested in what other things I can do with the plugin system.

**Comment by FabianKramm** [comment]
@agracey not sure if it helps, but we will also provide a new functionality for `vcluster connect` that might help you with this, where you could do something like `vcluster connect my-vcluster -n my-vcluster -- kubectl apply -f manifests.yaml` to connect and execute a command with the vcluster context at the same time.

Regarding the plugin functionality, we'll release a preview version early next week with several examples and documentation how to write those yourself.

**Comment by FabianKramm** [comment]
@agracey we just released version v0.6.0-alpha.7 that adds plugin support. For an example plugin, that is able to do what you want, take a look at the [bootstrap-with-deployment plugin](https://github.com/loft-sh/vcluster-sdk/tree/main/examples/bootstrap-with-deployment). We also have [docs](https://www.vcluster.com/docs/plugins/overview) that describe how the plugin system works and a complete walk through. We would be very grateful if you could provide us some feedback on this.

**Comment by agracey** [comment]
Thank you for the heads up. I'll try to get some time to code tomorrow or the weekend. 

I was reading the docs and just have to say that using sidecars to dynamically load plugins is a really clever pattern. I'm stealing that for my toolbox.

**Comment by olljanat** [comment]
Afaiu just released [v0.8.0](https://github.com/loft-sh/vcluster/releases/tag/v0.8.0) added support for this?

**Comment by FabianKramm** [comment]
@olljanat yes thats correct we added support for this in v0.8.0! I'm closing this
